### ISSUE TITLE
pulsar instrumentation not loaded by java agent

### ISSUE THREAD

**Issue by stevesea** [open-telemetry/opentelemetry-java-instrumentation#8372]
**Describe the bug**
It doesn't appear that Pulsar autoinstrumentation is working. When I enable javaagent debug logging, I don't see debug messages from the InstrumentationLoader indicating that it's loading the Pulsar 2.8 instrumentation.

**Steps to reproduce**
set OTEL_JAVAAGENT_DEBUG=true, observe the output does not include Pulsar

**What did you expect to see?**
The Pulsar instrumentation loaded, and  Messaging Spans created for pulsar consumer 

**What did you see instead?**
similar to jdbc and other instrumentations, I expected to see debug log messages reporting the pulsar instrumentation was loaded.

```
[otel.javaagent 2023-04-26 20:13:30:334 +0000] [main] DEBUG io.opentelemetry.javaagent.tooling.instrumentation.InstrumentationLoader - Loading instrumentation java-util-logging [class io.opentelemetry.javaagent.instrumentation.jul.JavaUtilLoggingInstrumentationModule]
[otel.javaagent 2023-04-26 20:13:30:335 +0000] [main] DEBUG io.opentelemetry.javaagent.tooling.instrumentation.InstrumentationLoader - Loading instrumentation jdbc [class io.opentelemetry.javaagent.instrumentation.jdbc.JdbcInstrumentationModule]
[otel.javaagent 2023-04-26 20:13:30:344 +0000] [main] DEBUG io.opentelemetry.javaagent.tooling.instrumentation.InstrumentationLoader - Loading instrumentation jdbc-datasource [class io.opentelemetry.javaagent.instrumentation.jdbc.datasource.DataSourceInstrumentationModule]
[otel.javaagent 2023-04-26 20:13:30:344 +0000] [main] DEBUG io.opentelemetry.javaagent.extension.instrumentation.InstrumentationModule - Instrumentation jdbc-datasource is disabled
```

setting the following env vars did not result in Pulsar instrumentation being loaded, or add messaging spans from my pulsar consumer.
```
OTEL_INSTRUMENTATION_PULSAR_ENABLED=true
OTEL_INSTRUMENTATION_PULSAR_EXPERIMENTAL_SPAN_ATTRIBUTES=true
OTEL_INSTRUMENTATION_MESSAGING_EXPERIMENTAL_RECEIVE_TELEMETRY_ENABLED=true
```

I do see some otel+pulsar logs from the javaagent -- messages like this from AgentInstaller$TransformLoggingListener: 
```
[otel.javaagent 2023-04-26 20:13:37:414 +0000] [main] DEBUG io.opentelemetry.javaagent.tooling.AgentInstaller$TransformLoggingListener - Transformed org.apache.pulsar.client.impl.ConnectionPool$$Lambda$755 -- jdk.internal.loader.ClassLoaders$AppClassLoader@39e67516
[otel.javaagent 2023-04-26 20:13:37:535 +0000] [pulsar-client-io-1-1] DEBUG io.opentelemetry.javaagent.tooling.AgentInstaller$TransformLoggingListener - Transformed org.apache.pulsar.common.protocol.ByteBufPair$Encoder -- jdk.internal.loader.ClassLoaders$AppClassLoader@39e67516
[otel.javaagent 2023-04-26 20:13:37:536 +0000] [pulsar-client-io-1-1] DEBUG io.opentelemetry.javaagent.tooling.AgentInstaller$TransformLoggingListener - Transformed org.apache.pulsar.common.protocol.ByteBufPair$CopyingEncoder -- jdk.internal.loader.ClassLoaders$AppClassLoader@39e67516
[otel.javaagent 2023-04-26 20:13:37:550 +0000] [pulsar-client-io-1-1] DEBUG io.opentelemetry.javaagent.instrumentation.executors.FutureInstrumentation - Skipping future instrumentation for org.apache.pulsar.client.util.TimedCompletableFuture
[otel.javaagent 2023-04-26 20:13:37:551 +0000] [pulsar-client-io-1-1] DEBUG io.opentelemetry.javaagent.tooling.AgentInstaller$TransformLoggingListener - Transformed org.apache.pulsar.client.util.TimedCompletableFuture -- jdk.internal.loader.ClassLoaders$AppClassLoader@39e67516
```

**What version are you using?**
v1.25.0

**Environment**
Compiler: Eclipse Adoptium OpenJDK 64-Bit Server VM 17.0.6+10
OS: eclipse-temurin:17.0.6_10-jre-jammy

**Additional context**
Added the v1.25.0 otel javaagent to backend service which has a Pulsar consumer.
Backend service is http4k, using Pulsar 2.11.1 client library. I was hoping to have new spans created via the Pulsar instrumentation.

set the following env vars
```
OTEL_INSTRUMENTATION_PULSAR_EXPERIMENTAL_SPAN_ATTRIBUTES=true
OTEL_INSTRUMENTATION_MESSAGING_EXPERIMENTAL_RECEIVE_TELEMETRY_ENABLED=true
```

The spans of the app just include jdbc, or the http library (undertow). and no spans from the Pulsar instrumentation.

**Comment by laurit** [comment]
It seems like by renaming the pulsar module in https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/8195 we accidentally didn't rename the subdirectory which results in the `pulsar` instrumentation being missing from javaagent.

**Comment by stevesea** [comment]
changed application to use the 1.24.0 javaagent, and confirmed that Pulsar Messaging Spans were exported.
Thank you!

**Comment by trask** [comment]
@stevesea thanks for reporting this! we have released 1.25.1 to fix this regression
